discovery-infra/start_discovery.py

```diff
@@ -169,6 +170,25 @@ def _create_node_details(cluster_name):
     }
 
 
+def validate_dns(client, cluster_id):
+    if args.managed_dns_domains == ":":
+        # 'set_dns' (using dnsmasq) is invoked after nodes_flow
+        return
+
+    cluster = client.cluster_get(cluster_id)
+    api_address = "api.{}.{}".format(cluster.name, cluster.base_dns_domain)
+    ingress_address = "ingress.apps.{}.{}".format(cluster.name, cluster.base_dns_domain)
+    try:
+        api_vip = socket.gethostbyname(api_address)
+        ingress_vip = socket.gethostbyname(ingress_address)
+        if api_vip == cluster.api_vip and ingress_vip == cluster.ingress_vip:
+            log.info("DNS domains are resolvable")
+        else:
+            raise Exception
```

Comment:
Exception("DNS domains are not resolvable") 
i except, lets add log.exception with relevant message

Reply:
and raise

Reply:
added